packages/wallet/src/redux/sagas/channel-locker.ts

```diff
@@ -0,0 +1,25 @@
+import { LockChannelRequest, channelLocked, channelUnlocked } from '../actions';
+import { take, takeEvery, put, call } from 'redux-saga/effects';
+
+export function* channelLocker() {
+  yield takeEvery('WALLET.LOCKING.LOCK_CHANNEL_REQUEST', lockChannel);
+}
+
+function* lockChannel(lockRequest: LockChannelRequest) {
+  let resolveLock;
+  const lockPromise = new Promise(resolve => (resolveLock = resolve));
+  (navigator as any).locks.request(
+    lockRequest.channelId,
+    { ifAvailable: true },
```

Comment:
`ifAvailable:true` means that attempting to acquire a lock on something that is already locked will return right away instead of blocking. We can use this to dispatch a `CHANNEL_BUSY` action so we can handle it in the protocol and show something to the user.